### PR TITLE
[FEATURE] add config for no Domain systems

### PR DESCRIPTION
**Pull request by mterwei**

### CODE REVIEW

File: Classes/Scheduler/Base.php [hunk 1]
```diff
@@ -171,9 +171,15 @@ public static function initFrontend( $pageid )
...
 		$GLOBALS['TSFE']->getConfigArray();
 
-		$rootline = BackendUtility::BEgetRootLine( $pageid );
-		if( ( $_SERVER['HTTP_HOST'] = BackendUtility::firstDomainRecord( $rootline ) ) == null ) {
-			throw new \RuntimeException( 'No domain record in root page' );
+		$rootline = BackendUtility::BEgetRootLine($pageid);
+		if ( BackendUtility::firstDomainRecord($rootline) != null ) {
+			$_SERVER['HTTP_HOST'] = BackendUtility::firstDomainRecord($rootline);
+		} elseif ( isset($GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['aimeos']['scheduler_domain']) ) {
+			$_SERVER['HTTP_HOST'] = $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['aimeos']['scheduler_domain'];
```
Comment (nos3) [inline Classes/Scheduler/Base.php]: EXTCONF is a bad idea for storing such data

Comment (aimeos) [reply Classes/Scheduler/Base.php]: Yes, it's not the best option

File: Classes/Scheduler/Base.php [hunk 2]
```diff
@@ -171,9 +171,15 @@ public static function initFrontend( $pageid )
...
+		} elseif ( !empty( getenv('SCHEDULER_DOMAIN' ) ) ) {
```
Comment (aimeos) [inline Classes/Scheduler/Base.php]: Should we use getenv('HTTP_HOST') to make it more clear?

**Comment by aimeos** [comment]
What about using that code:
```
$rootline = BackendUtility::BEgetRootLine($pageid);
if ( BackendUtility::firstDomainRecord($rootline) != null ) {
	$_SERVER['HTTP_HOST'] = BackendUtility::firstDomainRecord($rootline);
} elseif ( !empty( getenv('HTTP_HOST' ) ) ) {
	$_SERVER['HTTP_HOST'] = getenv('HTTP_HOST');
} else {
	throw new \RuntimeException('No config or domain record in root page found');
}
```

**Comment by aimeos** [comment]
If you change your PR accordingly, we will merge it into the master branch

**Comment by mterwei** [comment]
Ok can i use getenv('SCHEDULER_HTTP_HOST' ).   
i think getenv('HTTP_HOST' ) is set per default. so i have to override it to set my specific scheduler domain config

**Comment by maxmessing** [comment]
I recommend the version from mterwei - we already discussed several approaches but none of them fits for all cases, the one from mterwei does...

**Pull request merged**